main_window/device_manager.py

```diff
@@ -18,13 +18,16 @@ class DeviceType(Enum):
 
 class DeviceManager:
 
-    def __init__(self):
+    def __init__(self, required_versions: Dict[DeviceType, str], strict_versions=True):
+        self.required_versions = required_versions
+        self.strict_versions = strict_versions
+
         self._lock = RLock()
 
         self._device_type_to_device: Dict[DeviceType, RegisteredDevice] = dict()
         self._full_address_to_device: Dict[FullAddress, RegisteredDevice] = dict()
 
-    def register_device(self, device_type: DeviceType, full_address: FullAddress) -> None:
+    def register_device(self, device_type: DeviceType, device_version: str, full_address: FullAddress) -> None:
         with self._lock:
             if device_type not in self._device_type_to_device.keys() and full_address not in self._full_address_to_device.keys():
```

Comment:
pretty sure you could just do `device_type in self._device_type_to_device` (without `.keys()`), but that can be deferred to a small cleanup PR later - this PR has enough file changes as it is

Reply:
Good call. I just grepped for other offenders. Looks like ~20 across ~5 files. Will make a PR soon.